pex/tools/commands/virtualenv.py

```diff
@@ -219,6 +264,47 @@ def iter_executables(self):
             if is_exe(path):
                 yield path
 
+    def iter_distributions(self):
+        # type: () -> Iterator[DistributionInfo]
+        """"""
+        setuptools_path = tuple(third_party.expose(["setuptools"]))
+        _, stdout, _ = self.interpreter.execute(
+            args=[
+                "-c",
+                dedent(
+                    """\
+                    from __future__ import print_function
+
+                    import sys
+
+
+                    setuptools_path = {setuptools_path!r}
+                    sys.path.extend(setuptools_path)
+
+                    from pkg_resources import working_set
+
+
+                    for dist in working_set:
+                        if dist.location not in setuptools_path:
+                            print(
+                                "{{project_name}} {{version}} {{sys_path_entry}}".format(
+                                    project_name=dist.project_name,
+                                    version=dist.version,
+                                    sys_path_entry=dist.location,
+                                )
+                            )
+                    """.format(
+                        setuptools_path=setuptools_path
+                    )
+                ),
+            ]
+        )
+        for line in stdout.splitlines():
+            project_name, version, sys_path_entry = line.split()
```

Comment:
```suggestion
            project_name, version, sys_path_entry = line.split(maxsplit=2)
```
In case `sys_path_entry` contains spaces?

Reply:
Yup - good catch. Bug fixed with a previously failing test.